python/ray/serve/_private/deployment_state.py

```diff
@@ -2564,13 +2564,15 @@ def get_deployment_details(self, id: DeploymentID) -> Optional[DeploymentDetails
             )
 
     def get_deployment_statuses(
-        self, ids: List[DeploymentID] = None
+        self, ids: Optional[List[DeploymentID]] = None
     ) -> List[DeploymentStatusInfo]:
-        statuses = []
-        for id, state in self._deployment_states.items():
-            if not ids or id in ids:
-                statuses.append(state.curr_status_info)
-        return statuses
+        if ids is None:
+            return [
+                state.curr_status_info for state in self._deployment_states.values()
+            ]
+        else:
+            states = (self._deployment_states.get(id) for id in ids)
+            return [state.curr_status_info for state in states if state]
```

Comment:
Is there any reason not to change it to something like the following so we don't have to do 2 for loops? It would also improve readability I think.
```suggestion
            return [self._deployment_states[id].curr_status_info for id in ids if id in self._deployment_states]
```

Reply:
`if id in self._deployment_states` means we'd be doing a double-lookup/look-before-you-leap. In Python it's typically faster to just try it and handle the failure (especially if failures are rare), which is what the `.get()` pattern does.

Reply:
Got it, I'm not very familiar with the best way to hyper-optimize Python. If the lookup time is your concern, we can also do:
```
            statuses = []
            for id in ids:
                state = self._deployment_states.get(id)
                if state is not None:
                    statuses.append(state.curr_status_info)
            return statuses
```

Reply:
Sure!